### PR TITLE
Wdfn 221 median data step function

### PR DESCRIPTION
**Pull request by ayan-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/src/scripts/components/hydrograph/index.js [hunk 1]
```diff
@@ -182,42 +183,23 @@ const timeSeriesLegend = function(elem) {
...
-    }
+const plotMedianPoints = function (elem, {xscale, yscale, modulo, points}) {
+    let stepFunction = d3Line()
+        .curve(curveStepAfter)
+        .x(function(d) {
+            return xscale(d.dateTime);
+        })
+        .y(function(d) {
+            return yscale(d.value);
+        });
```
Comment (danielnaab) [inline assets/src/scripts/components/hydrograph/index.js]: That's pretty nice

File: assets/src/scripts/components/hydrograph/statistics.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        filtered.unshift(leftVal);
+    }
+    let last = filtered[filtered.length - 1];
+    if (last.dateTime < series.endTime) {
+        let rightVal = Object.assign({}, last);
+        rightVal.dateTime = series.endTime;
+        filtered.push(rightVal);
+    }
+    return filtered;
+};
```
Comment (danielnaab) [inline assets/src/scripts/components/hydrograph/statistics.js]: This function looks good, but could you add in a comment explaining each section of it?

**Comment by danielnaab** [comment]
I'm thinking that if we calculated time series-like data rather than store on the state, we should put the data outside of the `series` section of the state - which could be reserved for use of data from the IV service. Would that make sense? I'm not sure if that would cause some mixups on key references anywhere...

File: assets/src/scripts/components/hydrograph/statistics.js (hunk 2 above)
Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/statistics.js]: I need to run - will finish looking at this tomorrow.

**Comment by ayan-usgs** [comment]
> I'm thinking that if we calculated time series-like data rather than store on the state, we should put the data outside of the series section of the state - which could be reserved for use of data from the IV service.

I've been thinking that too. I would take on that task a separate ticket though because that refactor would likely ripple into other parts of the code base.

**Comment by mbucknell** [comment]
I agree with the discussion regarding moving the median data out of the series state. Another ticket sounds like the right thing to do.

**Pull request merged**